### PR TITLE
Searching algorithms

### PR DESCRIPTION
**Pull request by Vaibhavpratapsingh22**
All searching algoritms with C codes and README file which contains there pseudo code.

### CODE REVIEW

File: algorithms/searching/README.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                return(0)
+            [End of of statement]
+    Step6->Exit                
+
+### Complexity
+**Time complexity** - О(log n), where n is the number of items being sorted.
+
+**Space complexity** - O(1), due to auxillary space only.
+
+## 3.Breadth First Search
```
Comment (udeysingh95) [inline algorithms/searching/README.md:74]:
```suggestion
## 3. Breadth-First Search
```

File: algorithms/searching/README.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            [End of step 3 loop]
+    Step6-> [Finished]
+            Exit                    
+
+### Complexity
+**Time complexity** -The time complexity can be expressed as O(|V|+|E|), since every vertex and every edge will be explored in the worst case.|V| is the number of vertices and |E| is the number of edges in the graph. Note that O(|E|) may vary between O(1) and O(|V|^{2}), depending on how sparse the input graph is.
+
+**Space complexity** - O(|V|), where |V| is the cardinality of the set of vertices (as said before). If the graph is represented by an adjacency list it occupies O(|V|+|E|) space in memory, while an adjacency matrix representation occupies O(|V|^{2})            
+
+## 4. Depth First Search
```
Comment (udeysingh95) [inline algorithms/searching/README.md:107]:
```suggestion
## 4. Depth-First Search
```

File: algorithms/searching/README.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+              all the neighbours of node N.
+           [End of step 3 loop]
+    Step6->[Finished]
+          Exit
+
+
+## Complexity
+**Time complexity** - O(V+E), where V and E are number of vertices and edges.
+                   
+![edit2](https://img.shields.io/static/v1?label=topic&message=Algoritm&color=orange)![edit](https://img.shields.io/github/languages/top/vaibhavpratapsingh22/Courses)![edit2](https://img.shields.io/static/v1?label=madeby&message=Vaibhav&color=<COLOR>)![edit2](https://img.shields.io/static/v1?label=reviewer&message=Udey&color=<COLOR>)![edit](https://img.shields.io/static/v1?label=PRs&message=Welcome&color=<COLOR>)
```
Comment (udeysingh95) [inline algorithms/searching/README.md:145]: This is cool - how did you make it?

Comment (Vaibhavpratapsingh22) [reply algorithms/searching/README.md]: Source -https://shields.io/

**Pull request merged**